### PR TITLE
Sync Mobile User and Platform Accouts

### PR DESCRIPTION
**Pull request by BenjaminSsempala**
**_WHAT DOES THIS PR DO?_**
- Sync Mobile User and Platform Accouts

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - []()
- GitHub issues
    - Closes #<enter GitHub issue number here>

**_HOW DO I TEST OUT THIS PR?_**

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?_**

### CODE REVIEW

**Review by Baalmart — COMMENTED** [review]
Thanks @BenjaminSsempala for this.

1. Just curious about the previous endpoints relating to this which I created sometime back. Any new learnings that have come up over time?
2. Also, could you add some unit tests for this new functionality? Just to maintain the culture?

Thanks!

**Comment by BenjaminSsempala** [comment]
> Thanks @BenjaminSsempala for this.
> 
> 1. Just curious about the previous endpoints relating to this which I created sometime back. Any new learnings that have come up over time?
> 2. Also, could you add some unit tests for this new functionality? Just to maintain the culture?
> 
> Thanks!

Hey @Baalmart 
I initially started with trying to utilise the endpoints in the app but after careful analysis, I realised they might have been complicating the process unnecessarily.

- The Firebase signup Process in the mobile app has linked additional events for example logging different events, state updates and management etc. Additional modifications would also have to be made to these which could introduce breaking changes. Additionally, some events have to be triggered in-app and can't be triggered externally using the API.
- I realised since the signup or login process is already handled in the mobile app, what we needed was just functionality for adding the user to the platform.

**Comment by Baalmart** [comment]
> > Thanks @BenjaminSsempala for this.
> > 
> > 1. Just curious about the previous endpoints relating to this which I created sometime back. Any new learnings that have come up over time?
> > 2. Also, could you add some unit tests for this new functionality? Just to maintain the culture?
> > 
> > Thanks!
> 
> Hey @Baalmart I initially started with trying to utilise the endpoints in the app but after careful analysis, I realised they might have been complicating the process unnecessarily.
> 
> * The Firebase signup Process in the mobile app has linked additional events for example logging different events, state updates and management etc. Additional modifications would also have to be made to these which could introduce breaking changes. Additionally, some events have to be triggered in-app and can't be triggered externally using the API.
> * I realised since the signup or login process is already handled in the mobile app, what we needed was just functionality for adding the user to the platform.

Thanks @BenjaminSsempala for the clarifications.
With this new approach, how do we ensure that we do not get many "unreal" Platform accounts? 
As per the Firebase authentication process, are all the mobile app accounts `genuine` users?

File: src/auth-service/utils/test/ut_create-user.js [hunk 1]
```diff
@@ -2895,4 +2895,92 @@ describe("create-user-util", function () {
       });
     });
   });
+
+  describe("syncAnalyticsAndMobile", () => {
+    it("should create a new user when the user does not exist locally", async () => {
```
Comment (Baalmart) [inline src/auth-service/utils/test/ut_create-user.js:2900]: Thanks @BenjaminSsempala for this.
Awaiting for the tests for the corresponding `controller` and perhaps the routes as well but that can be ignored I guess.

Comment (BenjaminSsempala) [reply src/auth-service/utils/test/ut_create-user.js:2900]: Thanks @Baalmart 
I've added those

**Review by Baalmart — APPROVED** [review]
thanks @BenjaminSsempala

**Pull request merged**